sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/KafkaBundle.java

```diff
@@ -272,8 +282,17 @@ public <K, V> KafkaConsumer<K, V> createConsumer(
           org.apache.kafka.clients.consumer.ConsumerConfig.CLIENT_ID_CONFIG,
           consumerConfig.getClientId() + "-" + instanceId);
     }
+    KafkaConsumer<K, V> consumer =
+        new KafkaConsumer<>(consumerProperties, keyDeSerializer, valueDeSerializer);
+
+    if (micrometerConsumerListener != null) {
```

Comment:
It seems like the `micrometerConsumerLister` is created unconditionally in `run`. Shouldn't we fail if it is `null`?
In the [client bundle we also fail](https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-client-jersey/src/main/java/org/sdase/commons/client/jersey/JerseyClientBundle.java#L61) when the _create client_ calls are made too early.
Not sure if failing here would be breaking or if the code would fail anyways, when this is called before run.

However, in the case `micrometerConsumerListener` is `null` here, I assume, you won't get any metrics. Finding out why would be hard to find out. Would it make sense to log a warning in that case. At least it seems like the bundle API is used wrong by the consuming service, when `micrometerConsumerListener` is `null`.

Reply:
Good idea

add init check

```
if (!initialized) {
      throw new IllegalStateException("KafkaBundle not yet initialized!");
}
```

and logging

```
if (micrometerProducerListener == null) {
  LOGGER.error("MicrometerProducerListener is not initialized! Metrics will not be recorded.");
} else {
  micrometerProducerListener.producerAdded(clientId, producer);
}
```